go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -837,30 +883,24 @@ func (be *BuiltinBackupEngine) backupFile(ctx context.Context, params BackupPara
 	// optional pipe, tee, output file and hasher).
 	_, err = io.Copy(writer, reader)
 	if err != nil {
-		return vterrors.Wrap(err, "cannot copy data")
-	}
-
-	// Close gzip to flush it, after that all data is sent to writer.
-	if compressor != nil {
-		closeCompressorAt := time.Now()
-		if err = compressor.Close(); err != nil {
-			return vterrors.Wrap(err, "cannot close compressor")
-		}
-		params.Stats.Scope(stats.Operation("Compressor:Close")).TimedIncrement(time.Since(closeCompressorAt))
+		finalErr = vterrors.Wrap(err, "cannot copy data")
+		return finalErr
```

Comment:
Some comment as above and following such assignments and returns. Although I don't mind keeping it this way for readability.

Reply:
fixed